### PR TITLE
RBAC Manual

### PR DESCRIPTION
**Pull request by pradumnnec**

### CODE REVIEW

**Review by naveennec — CHANGES_REQUESTED** [review]

File: doc/en/source/k8sIntegration.rst [hunk 1]
```diff
@@ -457,3 +419,513 @@ Note: The value of **"namespace"** will be one which user specified while creati
...
 
 .. figure:: figures/microk8s_pods.png
+
+***********************
+Kubernetes Security 
+***********************
+
+Kubernetes provides many controls that can greatly improve an application's security. In order to use any of those methods provided by kubernetes, one need to properly configure the apiserver. **Role Based Access Control (RBAC)** is one such security implementation. RBAC is a method of regulating access to computer or network resources based on the roles of individual users within an organization. RBAC authorization uses the **rbac.authorization.k8s.io** API group to drive authorization decisions, allowing user to dynamically configure policies through the Kubernetes API.
+
+Implementing RBAC over Cloud Node Kubernetes Cluster
```
[the remark this answers is not in the document]
Comment (pradumnnec) [reply doc/en/source/k8sIntegration.rst]: Comment Incorporated

File: doc/en/source/k8sIntegration.rst [hunk 2]
```diff
@@ -457,3 +419,513 @@ Note: The value of **"namespace"** will be one which user specified while creati
...
+-----------------------------------------------------
+
+It is assumed that kubernetes cluter is setup and running at cloud node. Inorder to setup RBAC in cloud node download and extract cloud-chart as shown above. 
+
+
+Configuring values.yaml over Cloud Node Kubernetes Cluster
```
[the remark this answers is not in the document]
Comment (pradumnnec) [reply doc/en/source/k8sIntegration.rst]: Comment Incorporated

File: doc/en/source/k8sIntegration.rst [hunk 3]
```diff
@@ -457,3 +419,513 @@ Note: The value of **"namespace"** will be one which user specified while creati
...
+***********************
+Kubernetes Security 
+***********************
+
+Kubernetes provides many controls that can greatly improve an application's security. In order to use any of those methods provided by kubernetes, one need to properly configure the apiserver. **Role Based Access Control (RBAC)** is one such security implementation. RBAC is a method of regulating access to computer or network resources based on the roles of individual users within an organization. RBAC authorization uses the **rbac.authorization.k8s.io** API group to drive authorization decisions, allowing user to dynamically configure policies through the Kubernetes API.
+
+Implementing RBAC over Cloud Node Kubernetes Cluster
+-----------------------------------------------------
+
+It is assumed that kubernetes cluter is setup and running at cloud node. Inorder to setup RBAC in cloud node download and extract cloud-chart as shown above.
```
[the remark this answers is not in the document]
Comment (pradumnnec) [reply doc/en/source/k8sIntegration.rst]: Comment Incorporated

File: doc/en/source/k8sIntegration.rst [hunk 4]
```diff
@@ -457,3 +419,513 @@ Note: The value of **"namespace"** will be one which user specified while creati
...
+   # Specifies whether a service account should be created
+   create: true
+   # Annotations to add to the service account
+   annotations: {}
+   # The name of the service account to use.
+   # If not set and create is true, a name is generated using the fullname template
+   name: "fogflow-dns"   //CAN BE CHANGED AS PER USER'S NEED
+
+        
+- On deploying this chart using helm, the **namespace** with name **fogflow** is created and inside that a **sericeaccount** with name **fogflow-dns** is created. Once these namespace and serviceaccount is created, next roles and their rolebindings are created. The table lists the created roles and rolebinding.
```
[the remark this answers is not in the document]
Comment (pradumnnec) [reply doc/en/source/k8sIntegration.rst]: Comment Incorporated

File: doc/en/source/k8sIntegration.rst [hunk 5]
```diff
@@ -457,3 +419,513 @@ Note: The value of **"namespace"** will be one which user specified while creati
...
+|     Roles          |  RoleBindings  |    Scope             |
++--------------------+----------------+----------------------+
+| fogflow-root-role  |   RootUser     |  Cluster             |
++--------------------+----------------+----------------------+
+| fogflow-admin-role |   Admin        |  fogflow - namespace |
++--------------------+----------------+----------------------+
+| fogflow-user-role  |   EndUser      |  fogflow - namespace |
++--------------------+----------------+----------------------+
+
+- To verify the creation of above resources use following commands:
```
[the remark this answers is not in the document]
Comment (pradumnnec) [reply doc/en/source/k8sIntegration.rst]: Comment Incorporated

File: doc/en/source/k8sIntegration.rst [hunk 6]
```diff
@@ -457,3 +419,513 @@ Note: The value of **"namespace"** will be one which user specified while creati
...
+
+.. figure:: figures/ns.png
+
+.. code-block:: console
+
+   $kubectl get rolebindings --namespace=fogflow
+
+.. figure:: figures/rbaccloud.png
+
+Adding Users in Cloud Node Kubernetes Cluster
```
[the remark this answers is not in the document]
Comment (pradumnnec) [reply doc/en/source/k8sIntegration.rst]: Comment Incorporated

**Pull request closed without merging**